blog/15-02-2023-how-to-tokenize-physical-assets-on-celo/how-to-tokenize-physical-assets-on-celo.md

```diff
@@ -0,0 +1,300 @@
+
+*1.  // SPDX-License-Identifier:* MIT: This is a comment that specifies the license that the code is released under. In this case, it is the MIT license.
+    
+*2.  pragma solidity ^0.8.9;:* This is a Solidity compiler directive that specifies the version of the Solidity compiler to use. The ^ character means to use any version greater than or equal to 0.8.9, but less than 0.9.0.
+    
+*3.  import "@openzeppelin/contracts/token/ERC20/ERC20.sol";:* This is an import statement that brings in the ERC20 contract from the OpenZeppelin library. This contract provides the functionality to create an ERC20 token.
+    
+*4.  import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";:* This is an import statement that brings in the ERC20Burnable contract from the OpenZeppelin library. This contract provides the functionality to burn ERC20 tokens.
+    
+*5.  import "@openzeppelin/contracts/security/Pausable.sol";:* This is an import statement that brings in the Pausable contract from the OpenZeppelin library. This contract provides the functionality to pause and unpause the token contract.
+    
+*6.  import "@openzeppelin/contracts/access/Ownable.sol";:* This is an import statement that brings in the Ownable contract from the OpenZeppelin library. This contract provides the functionality to set an owner for the token contract.
+    
+*7.  contract TokenizationOfPhysicalAsset is ERC20, ERC20Burnable, Pausable, Ownable {:* This is the start of the main contract, which is named TokenizationOfPhysicalAsset. It inherits from the ERC20, ERC20Burnable, Pausable, and Ownable contracts that were imported earlier.
+    
+*8.  constructor() ERC20("Tokenization of Physical Asset", "Tokenize") {:* This is the constructor for the contract, which is called when the contract is first deployed. It initializes the token with a name of "Tokenization of Physical Asset" and a symbol of "Tokenize". The ERC20 contract's constructor is also called, which sets up some initial variables.
+    
+*9.  _mint(msg.sender, 100 * 10 ** decimals());:* This function call mints 100 tokens to the address that deployed the contract, which is represented by msg.sender. The number is multiplied by 10 ** decimals(), which sets the number of decimal places for the token. By default, the ERC20 standard has 18 decimal places, so this call will mint 100 tokens with 18 decimal places.
+    
+*10.  function pause() public onlyOwner {:* This function is used to pause the token contract. It can only be called by the contract owner.
+    
+*11.  _pause();:* This is the function call that actually pauses the token contract. It is provided by the Pausable contract.
+    
+*12.  function unpause() public onlyOwner {:* This function is used to unpause the token contract. It can only be called by the contract owner.
+    
+*13.  _unpause();:* This is the function call that actually unpauses the token contract. It is provided by the Pausable contract.
```

Comment:
*13. _unpause(); : * This is the function call that unpauses the token contract. It is provided by the Pausable contract.

Reply:
Done